README.md

```diff
@@ -135,13 +145,43 @@ Now create and apply the following manifest on your nxos devices.
   }
 ~~~
 
-## <a href='example-manifests'>Example Manifests</a>
+### <a name="setup-puppet-device">Puppet Device (Agentless)</a>
+
+The module supports remote management through the usage of [`puppet device`](https://puppet.com/docs/puppet/5.5/puppet_device.html).
+
+##### The `cisco_node_utils` Ruby Gem
+
+The [`cisco_node_utils`](https://rubygems.org/gems/cisco_node_utils) Ruby gem is a required component of the `ciscopuppet` module. This gem contains platform APIs for interfacing between Cisco CLI and Puppet resources. The gem will need installed on any Puppet agent which will be managing a NX-OS device. It can be automatically installed by Puppet by using the [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) helper class, or it can be installed manually.
+
+##### The `puppet-resource_api` Ruby Gem
+
+The [`puppet-resource_api`](https://rubygems.org/gems/puppet-resource_api) Ruby gem is a required component of the `ciscopuppet` module. The gem can be automatically installed by any Puppet agent which will be managing a device by simply using the [`ciscopuppet::install`](https://github.com/cisco/cisco-network-puppet-module/blob/master/examples/demo_all_cisco.pp#L19) helper class, or it can be installed manually.

```

Comment:
See above (and https://tickets.puppetlabs.com/browse/FM-7624) about gem installation.